src/hb-ot-layout-gpos-table.hh

```diff
@@ -566,6 +566,26 @@ struct AnchorMatrix
     return_trace (true);
   }
 
+  bool subset (hb_subset_context_t *c,
+	       unsigned cols,
+	       const hb_map_t *klass_mapping) const
+  {
+    TRACE_SUBSET (this);
+    auto *out = c->serializer->start_embed (*this);
+
+    auto indexes =
+    + hb_range (rows * cols)
+    | hb_filter ([=] (unsigned index) { return klass_mapping->has (index % cols); })
```

Comment:
There's an assumption in the output of `Markclass_closure_and_remap_indexes` that is not reflected on the type of the `hb_map_t *klass_map` but relied on in the code here and in other places, that basically the mapping is monotone and continuous. Let's commit this and work on improving internals later.

Reply:
Yeah I believe we frequently make that assumption for the glyph map as well. It might be nice to have a special version of map which enforces that.

Reply:
> Yeah I believe we frequently make that assumption for the glyph map as well. It might be nice to have a special version of map which enforces that.

Yep. We'll switch to a different data-structure. I think the CFF subsetter added something along those lines.

Reply:
> Yep. We'll switch to a different data-structure. I think the CFF subsetter added something along those lines.

hb_inc_bimap_t I think.